src/main/common/draft/draftStoreClient.ts

```diff
@@ -1,55 +1,67 @@
 import * as config from 'config'
-import * as HttpStatus from 'http-status-codes'
 import request from 'client/request'
 
+import { DraftDocument, Draft } from 'app/models/draft'
 import { MomentFactory } from 'common/momentFactory'
-import { Draft } from 'app/models/draft'
 
-const draftStoreConfig = config.get<any>('draft-store')
+const endpointURL: string = `${config.get<any>('draft-store').url}/drafts`
 
-if (draftStoreConfig.apiVersion !== 'v2') {
-  throw new Error(`Draft store API ${draftStoreConfig.apiVersion} is not supported by the client`)
-}
+export default class DraftStoreClient<T extends DraftDocument> {
+  private serviceAuthToken: string
 
-function withAuthHeader (userId: number, other: any = {}) {
-  return Object.assign({
-    headers: {
-      'Authorization': `hmcts-id ${userId}`
-    }
-  }, other)
-}
+  constructor (serviceAuthToken: string) {
+    this.serviceAuthToken = serviceAuthToken
+  }
 
-export default class DraftStoreClient<T> {
-  private endpointURL: string
+  find (query: { [key: string]: string }, userAuthToken: string, deserializationFn: (value: any) => T): Promise<Draft<T>[]> {
+    const { type, ...qs } = query
 
-  constructor (draftType: string) {
-    if (!draftType || draftType.trim() === '') {
-      throw new Error('Draft type is required by the client')
-    }
-    this.endpointURL = `${draftStoreConfig.url}/api/${draftStoreConfig.apiVersion}/draft/${draftType}`
+    return request
+      .get(endpointURL, {
+        qs: qs,
+        headers: this.authHeaders(userAuthToken)
+      })
+      .then((response: any) => {
+        return response.data
+          .filter(draft => type ? draft.type === type : true)
+          .map(draft => {
+            const wrapper = new Draft()
```

Comment:
I'm not sure if intelij can understand that `wrapper` is type of `Draft`?

maybe:

```
const wrapper:Draft = new Draft()
```

Reply:
it's pretty obvious ;) and intellij can understand it

Reply:
I decided not to give a time since you can see it at the same line. As a interesting thing a TSLint used by Angular is configured in such a way that it will return errors if you defining type in such an obvious cases.